### PR TITLE
Added additional helpers for enrollments

### PR DESCRIPTION
**Pull request by giocalitri**

### CODE REVIEW

File: edx_api/enrollments/models.py [hunk 1]
```diff
@@ -23,8 +28,8 @@ def enrolled_courses(self):
         """
         Returns a generator of all the courses the user has enrolled in
         """
-        for enrollment_json in self.payload:
-            yield Enrollment(enrollment_json)
+        for enrollment in self.enrollments.values():
```
Comment (justinabrahms) [inline edx_api/enrollments/models.py]: return self.enrollments.values()

Comment (giocalitri) [inline edx_api/enrollments/models.py]: for python 2.7 this will return a list

File: edx_api/enrollments/models.py [hunk 2]
```diff
@@ -39,25 +44,41 @@ def get_enrolled_courses_by_ids(self, course_id_list):
...
-        if course_id_list is not None:
-            enrollments = self.get_enrolled_courses_by_ids(course_id_list)
-        else:
-            enrollments = self.enrolled_courses
-        for enrollment in enrollments:
-            yield enrollment.course_id
+        all_ids = self.enrollments.keys()
+        if filter_by_id_list is not None:
+            all_ids = set(all_ids).intersection(set(filter_by_id_list))
+        for enrollmentcourse_id in all_ids:
```
Comment (justinabrahms) [inline edx_api/enrollments/models.py]: `return all_ids`

Comment (giocalitri) [inline edx_api/enrollments/models.py]: this will not be a generator...

**Comment by justinabrahms** [comment]
:+1:

**Pull request merged**